openedx/core/djangoapps/oauth_dispatch/tests/test_dot_overrides.py

```diff
@@ -71,3 +79,67 @@ def test_inactive_user_validates(self):
         self.user.save()
         request = self.request_factory.get('/')
         self.assertTrue(self.validator.validate_user('darkhelmet', '12345', client=None, request=request))
+
+
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+class CustomAuthorizationViewTestCase(TestCase):
+    """
+    Test custom authorization view works.
+
+    In particular, users should not be re-prompted to approve
+    an application even if the access token is expired.
+    (This is a temporary override until Auth Scopes is implemented.)
+    """
+    def setUp(self):
+        super(CustomAuthorizationViewTestCase, self).setUp()
+        self.user = User.objects.create_user(
+            username='darkhelmet',
+            password='12345',
+            email='darkhelmet@spaceball_one.org',
+        )
+        self.dot_adapter = adapters.DOTAdapter()
+        self.user = UserFactory()
+
+        self.restricted_dot_app = self.dot_adapter.create_confidential_client(
+            name='test restricted dot application',
+            user=self.user,
+            redirect_uri=DUMMY_REDIRECT_URL,
+            client_id='dot-restricted-app-client-id',
+        )
+        models.RestrictedApplication.objects.create(application=self.restricted_dot_app)
+
+    def _get_authorize(self, scope):
+        authorize_url = '/oauth2/authorize/'
+        return self.client.get(
+            authorize_url,
+            {
+                'client_id': self.restricted_dot_app.client_id,
+                'response_type': 'code',
+                'state': 'random_state_string',
+                'redirect_uri': DUMMY_REDIRECT_URL,
+                'scope': scope,
+            },
+        )
+
+    def test_no_reprompting_though_expired(self):
+        self.client.login(username=self.user.username, password='test')
+
+        date_in_the_past = timezone.now() + datetime.timedelta(days=-100)
+        dot_models.AccessToken.objects.create(
+            user=self.user,
+            token='1234567890',
+            application=self.restricted_dot_app,
+            expires=date_in_the_past,
+            scope='profile',
+        )
+
+        # redirects to redirect_uri without prompting, with previously approved scope
+        response = self._get_authorize(scope='profile')
+        self.assertEqual(response.status_code, 302)
+        self.assertTrue(response.url.startswith(DUMMY_REDIRECT_URL))
+
+        # returns a prompt when requesting a new scope
+        response = self._get_authorize(scope='email')
```

Comment:
Should these lines be a separate test?

Reply:
yes, that would be much better. will do.